tests/sanity/fake_cloud_provider.go

```diff
@@ -187,6 +187,15 @@ func (c *fakeCloudProvider) GetSnapshotByName(ctx context.Context, name string)
 	return snapshots[0].Snapshot, nil
 }
 
+func (c *fakeCloudProvider) GetSnapshotById(ctx context.Context, snapshotID string) (snapshot *cloud.Snapshot, err error) {
+	ret, exists := c.snapshots[snapshotID]
+	if !exists {
+		return nil, cloud.ErrNotFound
+	}
+
+	return ret.Snapshot, nil
+}
+
```

Comment:
It looks like you also have to change the `GetSnapshotByName` above, since it's comparing the the snapshot name with the ID.

Also, we should probably not use the snapshot ID as name in line 148 above.

Reply:
good catch. updated